nixos/modules/services/mail/cyrus-imap.nix

```diff
@@ -0,0 +1,366 @@
+                cmd = [ "notifyd" ];
+                listen = "/run/cyrus/notify";
+                proto = "udp";
+                prefork = 0;
+              };
+            };
+            description = ''
+              This section is the heart of the cyrus.conf file. It lists the processes that should be spawned to handle client connections made on certain Internet/UNIX sockets.
+            '';
+          };
+          EVENTS = mkOption {
+            default = {
+              tlsprune = {
+                cmd = [ "tls_prune" ];
+                at = 400;
+              };
+              delprune = {
+                cmd = [
+                  "cyr_expire"
+                  "-E"
+                  "3"
+                ];
+                at = 400;
+              };
+              deleteprune = {
+                cmd = [
+                  "cyr_expire"
+                  "-E"
+                  "4"
+                  "-D"
+                  "28"
+                ];
+                at = 430;
+              };
+              expungeprune = {
+                cmd = [
+                  "cyr_expire"
+                  "-E"
+                  "4"
+                  "-X"
+                  "28"
+                ];
+                at = 445;
+              };
+              checkpoint = {
+                cmd = [
+                  "ctl_cyrusdb"
+                  "-c"
+                ];
+                period = 30;
+              };
+            };
+            description = ''
+              This section lists processes that should be run at specific intervals, similar to cron jobs. This section is typically used to perform scheduled cleanup/maintenance.
+            '';
+          };
+          DAEMON = mkOption {
+            default = { };
+            description = ''
+              This section lists long running daemons to start before any SERVICES are spawned. master(8) will ensure that these processes are running, restarting any process which dies or forks. All listed processes will be shutdown when master(8) is exiting.
+            '';
+          };
+        };
+      };
+      description = "Cyrus configuration settings. [cyrus.conf(5)](https://www.cyrusimap.org/imap/reference/manpages/configs/cyrus.conf.html)";
+    };
+    imapdSettings = mkOption {
+      type = submodule {
+        freeformType = attrsOf (oneOf [
+          str
+          int
+          bool
+          (listOf str)
+        ]);
+        options = {
+          configdirectory = mkOption {
+            type = path;
+            default = "/var/lib/cyrus";
+            description = ''
+              The pathname of the IMAP configuration directory. This field is required.
+            '';
+          };
+          lmtpsocket = mkOption {
+            type = path;
+            default = "/run/cyrus/lmtp";
+            description = ''
+              Unix socket that lmtpd listens on, used by deliver(8). This should match the path specified in cyrus.conf(5).
```

Comment:
Can you change this to refer to a NixOS option?

Reply:
What if user don’t want to use it?
I think it can’t be set forcely.